app/src/main/java/com/globant/onboardingcalculatorwithkotlin/mvp/presenter/CalculatorPresenter.kt

```diff
@@ -62,17 +73,63 @@ class CalculatorPresenter(
             } else if (!model.first_operand.contains(DECIMAL_POINT)) {
                 model.first_operand = "${model.first_operand}$DECIMAL_POINT"
             } else {
-                view.showDecimalError()
+                view.showMessage(CalculatorError.DECIMAL_ERROR)
             }
         } else if (model.operator != EMPTY_CHAR) {
             if (model.second_operand.isEmpty()) {
                 model.second_operand = "$NUMBER_ZERO$DECIMAL_POINT"
             } else if (!model.second_operand.contains(DECIMAL_POINT)) {
                 model.second_operand = "${model.second_operand}$DECIMAL_POINT"
             } else {
-                view.showDecimalError()
+                view.showMessage(CalculatorError.DECIMAL_ERROR)
             }
         }
         updateVisor()
     }
+
+    private fun validOperation(): Boolean {
+        if (!model.second_operand.equals(NUMBER_ZERO))
```

Comment:
you can put here 

return !model.second_operand.equals(NUMBER_ZERO)

Reply:
Done!